truss-chains/examples/rag/rag_chain.py

```diff
@@ -0,0 +1,144 @@
+            chess in Central Park and exploring the city's diverse culinary scene.
+            """,
+            """
+            Sara Kim is a digital marketing specialist living in San Francisco.
+            She helps brands build their online presence with creative strategies.
+            Outside of work, Sara is passionate about photography and enjoys hiking
+            the trails around the Bay Area.
+            """,
+            """
+            David O'Connor calls New York City his home and works as a high school teacher.
+            He is dedicated to inspiring the next generation through education. In his free time,
+            David loves running along the Hudson River and participating in local theater productions.
+            """,
+            """
+            Lena Rossi is an architect based in San Francisco. She designs sustainable and
+            innovative buildings that contribute to the city's skyline. When she's not
+            working, Lena enjoys practicing yoga and exploring art galleries.
+            """,
+            """
+            Akio Tanaka lives in Tokyo and is a software developer specializing in mobile apps.
+            Akio is an avid gamer and enjoys attending eSports tournaments.
+            He also has a passion for cooking and often experiments with new recipes in his spare time.
+            """,
+            """
+            Maria Silva is a nurse residing in New York City. She is dedicated to providing
+            compassionate care to her patients. Maria finds joy in gardening and often
+            spends her weekends tending to her vibrant flower beds and vegetable garden.
+            """,
+            """
+            John Smith is a journalist based in San Francisco. He reports on international politics
+            and has a knack for uncovering compelling stories. Outside of work, John is a history
+            buff who enjoys visiting museums and historical sites.
+            """,
+            """
+            Aisha Mohammed lives in Tokyo and works as a graphic designer. She creates visually stunning
+            graphics for a variety of clients. Aisha loves to paint and often showcases her
+            artwork in local exhibitions.
+            """,
+            """
+            Carlos Mendes is an environmental engineer in San Francisco.
+            He is passionate about developing sustainable solutions for urban areas.
+            In his leisure time, Carlos enjoys surfing and participating in beach clean-up initiatives.
+            """,
+        ]
+
+        self._collection.add(documents=documents, ids=[f"id{n}" for n in range(10)])
+
+    async def run_remote(self, query: str) -> list[str]:
+        results = self._collection.query(
+            query_texts=[query],  # Chroma will embed this for you
+            n_results=2,  # how many results to return
+        )
+
+        if results is None or not results:
```

Comment:
Does this ever happen? Doesn't chroma just return the ones that are *relatively speaking* closest - even if they are not very close?